filter_linux.go

```diff
@@ -119,6 +120,148 @@ func (filter *Fw) Type() string {
 	return "fw"
 }
 
+type Flower struct {
+	FilterAttrs
+	DestIP        net.IP
+	DestIPMask    net.IPMask
+	SrcIP         net.IP
+	SrcIPMask     net.IPMask
+	EthType       uint16
+	EncDestIP     net.IP
+	EncDestIPMask net.IPMask
+	EncSrcIP      net.IP
+	EncSrcIPMask  net.IPMask
+	EncDestPort   uint16
+	EncKeyId      uint32
+
+	Actions []Action
+}
+
+func (filter *Flower) Attrs() *FilterAttrs {
+	return &filter.FilterAttrs
+}
+
+func (filter *Flower) Type() string {
+	return "flower"
+}
+
+func (filter *Flower) ipv4FullMask() net.IPMask {
+	return net.CIDRMask(32, 32)
+}
+
+func (filter *Flower) ipv6FullMask() net.IPMask {
+	return net.CIDRMask(128, 128)
+}
+
+func (filter *Flower) encodeIP(parent *nl.RtAttr, ip net.IP, mask net.IPMask, v4Type, v6Type int, v4MaskType, v6MaskType int) {
+	ipType := v4Type
+	maskType := v4MaskType
+
+	encodeMask := mask
+	if mask == nil {
+		encodeMask = filter.ipv4FullMask()
+	}
+	v4 := ip.To4()
+	if v4 == nil && len(ip) == net.IPv6len {
```

Comment:
`ip` is `net.IP`. So either it is an IPv4 (ip.To4() != nil) or it is an IPv6 (ip.To4() == nil). You should not need to check it is 16 bytes long.
Otherwise, on this logic, assume it is 14 bytes long. Now your code says "it is a IPv4".